libs/logger/logger_spdlog.cpp

```diff
@@ -13,16 +13,27 @@
 #include <spdlog/spdlog.h>
 #include <boost/assert.hpp>
 
-static const std::string kDefaultPattern =
-    R"([%Y-%m-%d %H:%M:%S.%F] [%L] [%n]: %v)";
+namespace {
 
-static const std::map<logger::LogLevel, const spdlog::level::level_enum>
-    kSpdLogLevels = {{logger::LogLevel::kTrace, spdlog::level::trace},
-                     {logger::LogLevel::kDebug, spdlog::level::debug},
-                     {logger::LogLevel::kInfo, spdlog::level::info},
-                     {logger::LogLevel::kWarn, spdlog::level::warn},
-                     {logger::LogLevel::kError, spdlog::level::err},
-                     {logger::LogLevel::kCritical, spdlog::level::critical}};
+  const std::string kDefaultPattern = R"([%Y-%m-%d %H:%M:%S.%F] [%L] [%n]: %v)";
+
+  spdlog::level::level_enum getSpdlogLogLevel(logger::LogLevel level) {
+    static const std::map<logger::LogLevel, const spdlog::level::level_enum>
+        kSpdLogLevels = {
+            {logger::LogLevel::kTrace, spdlog::level::trace},
+            {logger::LogLevel::kDebug, spdlog::level::debug},
+            {logger::LogLevel::kInfo, spdlog::level::info},
+            {logger::LogLevel::kWarn, spdlog::level::warn},
+            {logger::LogLevel::kError, spdlog::level::err},
+            {logger::LogLevel::kCritical, spdlog::level::critical}};
+    const auto it = kSpdLogLevels.find(level);
+    BOOST_ASSERT_MSG(it != kSpdLogLevels.end(), "Unknown log level!");
```

Comment:
Not sure we need to use asserts in logger that is going to be a part of everything.

Reply:
It will not affect the release build.